pkg/postgres/walker/stringutils.go

```diff
@@ -0,0 +1,43 @@
+package walker
+
+import (
+	"strings"
+
+	"github.com/stackrox/rox/pkg/stringutils"
+)
+
+func applyPointwise(ss []string, f func(string) string) {
+	for i, s := range ss {
+		ss[i] = f(s)
+	}
+}
+
+func makeSingular(s string) string {
+	l := len(s)
+	if l == 0 {
+		return s
+	}
+	if l == 1 && (s[0] == 's' || s[0] == 'S') {
+		return s[:l-1]
```

Comment:
```suggestion
		return ""
```
I think I don't get it. What this function does? If string is a single character it will return empty string?
It will be good to see some test for this logic

Reply:
Umm...should we be changing the string with one character? `s` or `S` cannot be determined as plural.

Reply:
I altered the logic as requested (empty or single-character strings are left untouched), added a method documentation comment and some basic tests for the function behaviour.